### ISSUE TITLE
Conflict with HTTPS Everywhere causes extension error warning

### ISSUE THREAD

**Issue by vith** [aawc/AlwaysSmileAmazon#7]
I still ended up at https://smile.amazon.com automatically in the end but I it caused the main Chrome 3 dot menu to turn into an orange warning icon to tell me about it, so I figure it's still worth reporting.

The error was like this:

> Warning:
> This extension failed to redirect a network request to http://smile.amazon.com/dp/asdf/ref=fdsa because another extension (HTTPS Everywhere) redirected it to https://www.amazon.com/dp/asdf/ref=fdsa.

Unfortunately I don't know how to reproduce it. It may have happened when I opened an old order from Google Inbox but I don't think I noticed it right away. Retracing my steps I don't get another error.

**Comment by aawc** [comment]
Thanks for reporting this issue and sorry for the delay in responding.

Unfortunately, there's nothing that this extension (or any other) can do to avoid it since this is working as intended from Chrome's Extension API perspective. See https://developer.chrome.com/extensions/webRequest#conflict_resolution

> If more than one extension attempts to modify the request, the most recently installed extension wins and all others are ignored. An extension is not notified if its instruction to modify or redirect has been ignored.

To be fair to the API authors, it's a reasonable solution to a problem that has no clear solution.

**Comment by vith** [comment]
OK, fair enough. I thought it might be possible to cancel the event but I don't see anything like that in the docs either.

Unfortunately I just had it happen again and this time I ended up at https://www.amazon.com rather than smile, so this might warrant further investigation.

From this page http://www.gamersnexus.net/hwreviews/2767-antlion-audio-modmic-5-review-quality-comparison I clicked the link with the text `Samson Meteor ($66)`. This one seems to happen consistently: 3 tries in a row all with the same result. Another difference with this case for me is I have to click through a uBlock Origin interstital because it's blocked by default as a monetization link.

The extension error on the Chrome menu only appeared the first time though. After the 3 tests, I tried clicking your extension's icon, which first only reloaded the same page (`www.`); the second time I clicked it I finally got to `smile.`
